File: qqbot-runtime/src/main/java/com/mieai/qqbot/runtime/outbox/OutboundTextPayload.java
package com.mieai.qqbot.runtime.outbox;

import com.mieai.qqbot.client.QqMessageTargetType;

/** Durable text-message contract consumed by the production Outbox worker. */
public record OutboundTextPayload(
        QqMessageTargetType targetType,
        String targetId,
        String content,
        String replyMessageId,
        String replyEventId,
        int messageSequence) {
    public static final String JOB_TYPE = "QQ_SEND_TEXT";
}

File: qqbot-runtime/src/main/kotlin/com/mieai/qqbot/runtime/outbox/OutboundTextPayload.kt
package com.mieai.qqbot.runtime.outbox

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.mieai.qqbot.client.QqMessageTargetType

data class OutboundTextPayload @JsonCreator(mode = JsonCreator.Mode.PROPERTIES) constructor(
    @JsonProperty("targetType") val targetType: QqMessageTargetType,
    @JsonProperty("targetId") val targetId: String,
    @JsonProperty("content") val content: String,
    @JsonProperty("replyMessageId") val replyMessageId: String?,
    @JsonProperty("replyEventId") val replyEventId: String?,
    @JsonProperty("messageSequence") val messageSequence: Int,
) {
    companion object {
        const val JOB_TYPE = "QQ_SEND_TEXT"
    }
}
